platform/server-bundle/src/main/resources/webapp/js/services.js

```diff
@@ -2,9 +2,277 @@
+                btnOKLabel: BootstrapDialog.DEFAULT_TEXTS.OK,
+                btnOKClass: null,
+                callback: null
+            };
+            if (typeof arguments[0] === 'object' && arguments[0].constructor === {}.constructor) {
+                options = $.extend(true, defaultOptions, arguments[0]);
+            } else {
+                options = $.extend(true, defaultOptions, {
+                    message: arguments[0],
+                    closable: false,
+                    buttonLabel: BootstrapDialog.DEFAULT_TEXTS.OK,
+                    callback: typeof arguments[1] !== 'undefined' ? arguments[1] : null
+                });
+            }
+            if (options.btnOKClass === null) {
+                options.btnOKClass = ['btn', options.type.split('-')[1]].join('-');
+            }
+
+            dialog = new BootstrapDialog({
+                type: options.type,
+                title: options.title,
+                message: options.message,
+                closable: options.closable,
+                draggable: options.draggable,
+                data: {
+                    callback: options.callback
+                },
+                buttons: [{
+                    label: options.btnCancelLabel,
+                    action: function (dialog) {
+                        BootstrapDialogManager.close(dialog);
+                        if (typeof dialog.getData('callback') === 'function' && dialog.getData('callback').call(this, false) === false) {
+                            return false;
+                        }
+                    }
+                }, {
+                    label: options.btnOKLabel,
+                    cssClass: options.btnOKClass,
+                    action: function (dialog) {
+                        BootstrapDialogManager.close(dialog);
+                        if (typeof dialog.getData('callback') === 'function' && dialog.getData('callback').call(this, true) === false) {
+                            return false;
+                        }
+                    }
+                }]
+            });
+            BootstrapDialogManager.open(dialog, false);
+        };
+
+        this.openLoadingModal = function () {
+            LoadingModal.open();
+        };
+
+        this.closeLoadingModal = function () {
+            LoadingModal.close();
+        };
+
+        this.motechAlert = function (msg, title, params, callback) {
+            this.alert({
+                title: jQuery.i18n.prop(title),
+                message: jQuery.i18n.prop.apply(null, [msg].concat(params)),
+                callback: callback
+            });
+        };
+
+        this.motechAlertStackTrace = function (msg, title, response, callback) {
+            if( title === null || title === '') {
+                title = 'Alert';
+            }
+            this.alert({
+                title: title,
+                message: jQuery.i18n.prop(msg).bold() + ": \n" + response,
```

Comment:
Let's not put formatting into the code directly. 

Can we use $compile here and render a template? This would allow us to use a directive (or something) to format the stack trace (in the future eventually) without touching the javascript...